Crashlytics does not start with .NET 8

I've recently upgraded a project where Crashlytics was working fine.

Now when I run the project I am getting following errors:

```
[I-CLS000000] Crashlytics could not find the symbol for the app's main function and cannot start up. This can happen when Exported Symbols File is set in Build Settings. To resolve this, add "__mh_execute_header" as a newline to your Exported Symbols File.

[I-CLS000000] Crash reporting could not be initialized
```

I double checked that strip symbols option was not enabled also tried adding ` <DebugType>full</DebugType>` but still the same issue.

Any idea on what might be the issue ?

> @claudioredi Thank you! I was able to fix the issue following your comment. I had to change it a bit to adapt it to Maui
> 
> Like you said, I created a symbols.list (I named it exported_symbols.txt) and added it to the folder Platforms/iOS
> 
> ```
> <ItemGroup Condition="'$(TargetFramework)' == 'net8.0-ios'">
>     <_ReferencesLinkerFlags Include="-u__mh_execute_header" Visible="false" />
>     <_CustomLinkFlags Include="-exported_symbols_list" Visible="false" />
>     <_CustomLinkFlags Include="$(ProjectDir)Platforms/iOS/exported_symbols.txt" Visible="false" />
> </ItemGroup>
> ```

Thank you for your comment. I have tried this workaround and it works in debug mode, but when I generate the artifact in release mode I get this error:

```
    Undefined symbols for architecture arm64:
      "﻿__mh_execute_header", referenced from:
         -exported_symbol[s_list] command line option
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

 Any idea how to fix it?

--- a/docs/bundled.md
+++ b/docs/bundled.md
@@ -20,6 +20,7 @@ If you encounter a build error, try to add the package via `dotnet add package P
     ...
 </resources>
 ```
+- For Crashlytics troubleshooting, including Android resource setup and iOS `__mh_execute_header` setup, see the [Crashlytics documentation](crashlytics.md).
 - Add the following line of code to the place where your app gets bootstrapped:
 ```c#
 #if IOS

--- a/docs/crashlytics.md
+++ b/docs/crashlytics.md
@@ -17,7 +17,29 @@ Firebase [Firebase Crashlytics](https://firebase.google.com/docs/crashlytics) is
 ```
 
 ### iOS specifics
-- Make sure to start your app without the debugger attached to be able to see the crashes in the firebase console
+- Make sure to start your app without the debugger attached when testing fatal crashes.
+- After forcing a crash, restart the app so Crashlytics can send the report. The crash may take several minutes to appear in the Firebase Console.
+- To enable verbose Firebase logs while diagnosing setup issues, add `-FIRDebugEnabled` to your iOS app launch arguments.
+- If the device log contains `Crashlytics could not find the symbol for the app's main function`, add the following properties to your app project:
+```xml
+<PropertyGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">
+    <_ExportSymbolsExplicitly>false</_ExportSymbolsExplicitly>
+    <DebugType>full</DebugType>
+</PropertyGroup>
+```
+- If the symbol error still appears, create `Platforms/iOS/exported_symbols.txt` containing exactly:
+```text
+__mh_execute_header
+```
+Then reference that file from your app project:
+```xml
+<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">
+    <_ReferencesLinkerFlags Include="-u__mh_execute_header" Visible="false" />
+    <_CustomLinkFlags Include="-exported_symbols_list" Visible="false" />
+    <_CustomLinkFlags Include="$(ProjectDir)Platforms/iOS/exported_symbols.txt" Visible="false" />
+</ItemGroup>
+```
+The exported symbols file must be plain UTF-8 without a byte order mark or other hidden characters. A hidden character before `__mh_execute_header` can cause Release builds to fail during linking.
 - For more specific instructions take a look at the official [Firebase documentation](https://firebase.google.com/docs/crashlytics/get-started?platform=ios)
 
 ### Android specifics
@@ -30,11 +52,18 @@ Firebase [Firebase Crashlytics](https://firebase.google.com/docs/crashlytics) is
     ...
 </resources>
 ```
+- If you created `strings.xml` manually, make sure it is included as an Android resource. In Visual Studio this is the `AndroidResource` build action. If the file is not already included in your project file, add an item like:
+```xml
+<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">
+    <AndroidResource Include="Platforms/Android/Resources/values/strings.xml" />
+</ItemGroup>
+```
+- After adding or changing Android resources, clean `bin` and `obj` and rebuild the app if Crashlytics still reports a missing build ID.
 - For more specific instructions take a look at the official [Firebase documentation](https://firebase.google.com/docs/crashlytics/get-started?platform=android)
 
 ## Usage
 
-To test if everything is setup correctly, restart the app after a forced crash and visit the [Crashlytics Dashboard](https://console.firebase.google.com/u/0/project/_/crashlytics) to view your reports and statistics.
+To test if everything is setup correctly, restart the app after a forced crash and visit the [Crashlytics Dashboard](https://console.firebase.google.com/u/0/project/_/crashlytics) to view your reports and statistics. See the official Firebase [test implementation guide](https://firebase.google.com/docs/crashlytics/test-implementation?platform=ios) for current platform-specific testing guidance.
 
 Take a look at the [documentation](https://github.com/AdamEssenmacher/GoogleApisForiOSComponents/blob/master/docs/Firebase/Crashlytics/GettingStarted.md) for the AdamE.Firebase.iOS.Crashlytics packages, because Plugin.Firebase's code is abstracted but still very similar.
 

